clang/lib/CodeGen/Targets/X86.cpp

```diff
@@ -792,7 +792,7 @@ ABIArgInfo X86_32ABIInfo::classifyArgumentType(QualType Ty, CCState &State,
         return ABIArgInfo::getDirect();
       return ABIArgInfo::getExpand();
     }
-    return getIndirectResult(Ty, /*ByVal=*/false, State);
+    return getIndirectResult(Ty, IsVectorCall && Ty->isBuiltinType(), State);
```

Comment:
Maybe try to avoid byval overhead:

```
if (IsVectorCall && Ty->isBuiltinType())
  return ABIArgInfo::getDirect();
return getIndirectResult(Ty, /*ByVal=*/false, State);
```